src/js/view/floating/floating-drag-ui.js

```diff
@@ -9,52 +10,58 @@ export default class FloatingDragUI {
 
     disable() {
         if (this.ui) {
+            // 'Dragged' state is reset so the transition animation can fire again if the player re-floats.
+            removeClass(this.container, 'jw-floating-dragged');
+            removeClass(this.container, 'jw-floating-dragging');
             this.ui.destroy();
             this.ui = null;
         }
     }
 
     enable() {
-        let playerLeft;
-        let playerTop;
-        let innerHeight;
-        let innerWidth;
-        const auto = 'auto';
         const { container, input } = this;
-        const ui = this.ui = new UI(input, { preventScrolling: true })
-            .on('dragStart', () => {
-                playerLeft = container.offsetLeft;
-                playerTop = container.offsetTop;
-                innerHeight = window.innerHeight;
-                innerWidth = window.innerWidth;
+        let startX;
+        let startY;
+        let deltaX;
+        let deltaY;
+        let x = 0;
+        let y = 0;
+        // A min/max delta is assigned to prevent the player from being dragged off screen.
+        let minDeltaX;
+        let minDeltaY;
+        let maxDeltaX;
+        let maxDeltaY;
+        this.ui = new UI(input, { preventScrolling: true })
+            .on('dragStart', (e) => {
+                const { pageX, pageY } = e;
+                const { outerWidth, outerHeight } = window;
+                const { offsetLeft, offsetTop, offsetWidth, offsetHeight } = container;
+                startX = pageX;
+                startY = pageY;
+                minDeltaX = calculateMin(offsetLeft);
+                minDeltaY = calculateMin(offsetTop);
```

Comment:
`calculateMin` just negates values. There's an operator for that (`-`):
```suggestion
                minDeltaX = -offsetLeft;
                minDeltaY = -offsetTop;
```

Reply:
Good point, was overthinking here. Making the change now